mcs_benchmark_data/_benchmark_transformer.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import json
+import xmltodict
+from abc import abstractmethod
+from pathlib import Path
+from rdflib import URIRef
+from typing import Generator
+
+from mcs_benchmark_data._model import _Model
+from mcs_benchmark_data._transformer import _Transformer
+
+from mcs_benchmark_data.models.benchmark import Benchmark
+from mcs_benchmark_data.models.benchmark_bootstrap import BenchmarkBootstrap
+from mcs_benchmark_data.models.benchmark_sample import BenchmarkSample
+from mcs_benchmark_data.models.benchmark_question_type import BenchmarkQuestionType
+from mcs_benchmark_data.models.benchmark_train_dataset import BenchmarkTrainDataset
+from mcs_benchmark_data.models.benchmark_test_dataset import BenchmarkTestDataset
+from mcs_benchmark_data.models.benchmark_dev_dataset import BenchmarkDevDataset
+
+
+class _Benchmark_Transformer(_Transformer):
+    """
+    Abstract base class for transformers.
+    See the transform method.
+    """
+
+    @property
+    def _uri_base(self):
+        return f"benchmark:{self._pipeline_id}"
+
+    @property
+    def _benchmark_dataset_classes(self):
+        return {
+            "dev": BenchmarkDevDataset,
+            "test": BenchmarkTestDataset,
+            "train": BenchmarkTrainDataset,
+        }
+
+    def _transform(self, **kwds) -> Generator[_Model, None, None]:
+
+        benchmark_json_file_path = kwds["extracted_path"] / getattr(
+            kwds["file_names"], "meta_data"
+        )
+
+        with open(benchmark_json_file_path) as benchmark_json:
+            benchmark_metadata = json.loads(benchmark_json.read())
+
+        benchmark_bootstrap = BenchmarkBootstrap.from_dict(benchmark_metadata)

```

Comment:
Might call this dataclass BenchmarkMetadata instead of BenchmarkBootstrap, so the names line up. Fine to leave for another PR.